### ISSUE TITLE
eval的结果miou只有1.4好像，请求指导

### ISSUE THREAD

**Issue by ksama777** [charlesChen02/RandLA-pytorch#5]
您好，我把checkpoint的路径设置成您那个pretrain文件夹里面的checkpoint ，test切片进行， sequence08的eval的结果好像有些不对 您能帮我看看吗 感谢您  

这是我test时候的命令 python3 test_SemanticKITTI.py --test_id 08 --result_dir /media/wx/HDD/DQ/RandLA-pytorch-main/result

(randlanet) wx@dl-group-workstation:/media/wx/HDD/DQ/RandLA-pytorch-main$ python3 evaluate_SemanticKITTI.py --dataset /media/wx/HDD/DQ/semantic_kitti/dataset/sequences --predictions /media/wx/HDD/DQ/RandLA-pytorch-main/result --sequence 08
********************************************************************************
INTERFACE:
Eval What: all
Data:  /media/wx/HDD/DQ/semantic_kitti/dataset/sequences
Predictions:  /media/wx/HDD/DQ/RandLA-pytorch-main/result
Sequences:  ['08']
Config:  /media/wx/HDD/DQ/RandLA-pytorch-main/utils/semantic-kitti.yaml
Limit:  None
********************************************************************************
Opening data config file /media/wx/HDD/DQ/RandLA-pytorch-main/utils/semantic-kitti.yaml
Ignoring xentropy class  0  in IoU evaluation
[IOU EVAL] IGNORE:  [0]
[IOU EVAL] INCLUDE:  [ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]
4071
4071 4071
Evaluating sequences
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4071/4071 [02:13<00:00, 30.40it/s]
Validation set:
Acc avg 0.063
IoU avg 0.014
IoU class 1 [car] = 0.017
IoU class 2 [bicycle] = 0.001
IoU class 3 [motorcycle] = 0.000
IoU class 4 [truck] = 0.000
IoU class 5 [other-vehicle] = 0.004
IoU class 6 [person] = 0.001
IoU class 7 [bicyclist] = 0.002
IoU class 8 [motorcyclist] = 0.000
IoU class 9 [road] = 0.090
IoU class 10 [parking] = 0.001
IoU class 11 [sidewalk] = 0.046
IoU class 12 [other-ground] = 0.001
IoU class 13 [building] = 0.009
IoU class 14 [fence] = 0.000
IoU class 15 [vegetation] = 0.012
IoU class 16 [trunk] = 0.000
IoU class 17 [terrain] = 0.079
IoU class 18 [pole] = 0.001
IoU class 19 [traffic-sign] = 0.000
********************************************************************************
0.017,0.001,0.000,0.000,0.004,0.001,0.002,0.000,0.090,0.001,0.046,0.001,0.009,0.000,0.012,0.000,0.079,0.001,0.000,0.014,0.063

**Comment by charlesChen02** [comment]
请指定test命令的--checkpoint_path 为 pretrain_model/checkpoint.tar 再测试

**Comment by ksama777** [comment]
您好，用了pretrain的checkpoint&nbsp;结果如下&nbsp;


这么看来是我训练的时候的问题？


(randlanet) ***@***.***:/media/wx/HDD/DQ/RandLA-pytorch-main$ python3 evaluate_SemanticKITTI.py --dataset /media/wx/HDD/DQ/semantic_kitti/dataset/sequences --predictions /media/wx/HDD/DQ/RandLA-pytorch-main/result --sequence 08
********************************************************************************
INTERFACE:
Eval What: all
Data:&nbsp; /media/wx/HDD/DQ/semantic_kitti/dataset/sequences
Predictions:&nbsp; /media/wx/HDD/DQ/RandLA-pytorch-main/result
Sequences:&nbsp; ['08']
Config:&nbsp; /media/wx/HDD/DQ/RandLA-pytorch-main/utils/semantic-kitti.yaml
Limit:&nbsp; None
********************************************************************************
Opening data config file /media/wx/HDD/DQ/RandLA-pytorch-main/utils/semantic-kitti.yaml
Ignoring xentropy class&nbsp; 0&nbsp; in IoU evaluation
[IOU EVAL] IGNORE:&nbsp; [0]
[IOU EVAL] INCLUDE:&nbsp; [ 1&nbsp; 2&nbsp; 3&nbsp; 4&nbsp; 5&nbsp; 6&nbsp; 7&nbsp; 8&nbsp; 9 10 11 12 13 14 15 16 17 18 19]
4071
4071 4071
Evaluating sequences
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4071/4071 [01:44<00:00, 38.97it/s]
Validation set:
Acc avg 0.903
IoU avg 0.550
IoU class 1 [car] = 0.940
IoU class 2 [bicycle] = 0.091
IoU class 3 [motorcycle] = 0.340
IoU class 4 [truck] = 0.666
IoU class 5 [other-vehicle] = 0.460
IoU class 6 [person] = 0.553
IoU class 7 [bicyclist] = 0.687
IoU class 8 [motorcyclist] = 0.000
IoU class 9 [road] = 0.920
IoU class 10 [parking] = 0.401
IoU class 11 [sidewalk] = 0.784
IoU class 12 [other-ground] = 0.005
IoU class 13 [building] = 0.886
IoU class 14 [fence] = 0.519
IoU class 15 [vegetation] = 0.855
IoU class 16 [trunk] = 0.626
IoU class 17 [terrain] = 0.748
IoU class 18 [pole] = 0.567
IoU class 19 [traffic-sign] = 0.399
********************************************************************************
0.940,0.091,0.340,0.666,0.460,0.553,0.687,0.000,0.920,0.401,0.784,0.005,0.886,0.519,0.855,0.626,0.748,0.567,0.399,0.550,0.903





ksama
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "charlesChen02/RandLA-pytorch"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月11日(星期二) 下午4:26
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [charlesChen02/RandLA-pytorch] eval的结果miou只有1.4好像，请求指导 (Issue #5)





 
请指定test命令的--checkpoint_path 为 pretrain_model/checkpoint.tar 再测试

**Comment by charlesChen02** [comment]
这么看整个模型是没啥问题的,  关于训练, 可以建议几点:
1. batch size尽量调大至最大化利用GPU资源, 可以缩短整体训练时间, 提高效率, 
2. epoch 尽量多训练一些, pretrain model是训练的80个左右得到的, 在V100大概用了一整天来训练
3. utils/config.py 中有设定 `num_points = 4096 * 10  # Number of input points` 代表了一个receptive field 的大小, 一般来说越大效果越好. 如果你在非kitti数据集上训练的话记得调整
